orbital_core/utils/tracing.py

```diff
@@ -0,0 +1,149 @@
+import sys
+import aiotask_context
+import aiozipkin
+import logging
+from traceback import format_exception
+from typing import Set
+from aiohttp.web import HTTPException, Application, HTTPInternalServerError, Request
+from aiozipkin import APP_AIOZIPKIN_KEY, REQUEST_AIOZIPKIN_KEY
+from aiozipkin.constants import HTTP_STATUS_CODE, HTTP_ROUTE
+from aiozipkin.helpers import TRACE_ID_HEADER, SPAN_ID_HEADER, PARENT_ID_HEADER, \
+    make_context, parse_sampled, parse_debug
+from aiozipkin.record import Record
+from aiozipkin.span import SpanAbc
+from aiozipkin.tracer import Tracer
+from aiozipkin.transport import Transport
+
+LOG = logging.getLogger(__name__)
+
+
+class TracingTransport(Transport):
+
+    def __init__(self, *args, **kargs):
+        pass
+
+    def send(self, record: Record) -> None:
+        extra = {
+            "span": self._record_to_key(record),
+            "topic_prefix": "tra",
+            "log_format": "span-g",
+        }
+        LOG.info("", extra=extra)
+
+    async def close(self) -> None:
+        pass
+
+    def _record_to_key(self, record: Record):
+        data = record.asdict()
+        return {
+            "xti": data["traceId"],
+            "xsi": data["id"],
+            "xpi": data["parentId"],
+            "xsb": '1' if record._context.sampled else '0', # Sampled
+            "xfl": data["debug"], # Flags
+            "spn": data["tags"].get(HTTP_ROUTE, data["name"].split(" ")[1]),
+            "spt": data["tags"],
+            "tsu": data["timestamp"], # Start Timestamp
+            "lat": data["duration"], # Latency of trace
+            "spk": data["kind"],
+        }
+
+
+def configure_tracing(app: Application, route_blacklist: Set[str]):
+    # aiocontext is how tracing context is pinned per
+    # coroutine, so it must be set.
+    # app.loop is not accessible before startup. Setting it as a
+    # startup task ensures that it will exist before a call occurs.
+    app.on_startup.append(_add_context_task_factory)
+    sampler = aiozipkin.Sampler(sample_rate=1.0)
```

Comment:
does this mean that every trace is sent? it's interesting that there is client-side sampling here... I feel like that might be an antipattern, but maybe that's the right choice. @almonteb thoughts?

Reply:
since we do downsampling out of band internally @ Z we opt to sample 100% from the client. in the public, we should probably make this configurable via `bootstrap_app`

Reply:
maybe let's expose it if people ask. I'm fine with the opinion being that sampling should happen outside the app.

Reply:
I have added a provision to accept sample rate.